include/alpaka/core/BoostPredef.hpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * \file
+ * Copyright 2017 Alexander Matthes
+ *
+ * This file is part of alpaka.
+ *
+ * alpaka is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * alpaka is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with alpaka.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <boost/predef.h>
+
+//-----------------------------------------------------------------------------
+// In boost since 1.68.0
+// BOOST_PREDEF_MAKE_10_VVRRP(V)
+#if !defined(BOOST_PREDEF_MAKE_10_VVRRP)
+    #define BOOST_PREDEF_MAKE_10_VVRRP(V) BOOST_VERSION_NUMBER(((V)/1000)%100,((V)/10)%100,(V)%10)
+#endif
+
+//-----------------------------------------------------------------------------
+// In boost since 1.68.0
+// CUDA language detection
+// - clang defines __CUDA__ and __CUDACC__ when compiling CUDA code ('-x cuda')
+// - nvcc defines __CUDACC__ when compiling CUDA code
+#if !defined(BOOST_LANG_CUDA)
+    #if defined(__CUDA__) || defined(__CUDACC__)
+        #include <cuda.h>
+        #define BOOST_LANG_CUDA BOOST_PREDEF_MAKE_10_VVRRP(CUDA_VERSION)
+    #else
+        #define BOOST_LANG_CUDA BOOST_VERSION_NUMBER_NOT_AVAILABLE
+    #endif
+#endif
+
+//-----------------------------------------------------------------------------
+// In boost since 1.68.0
+// CUDA device architecture detection
+#if !defined(BOOST_ARCH_PTX)
+    #if defined(__CUDA_ARCH__)
+        #define BOOST_ARCH_PTX BOOST_PREDEF_MAKE_10_VRP(__CUDA_ARCH__)
+    #else
+        #define BOOST_ARCH_PTX BOOST_VERSION_NUMBER_NOT_AVAILABLE
+    #endif
+#endif
+
+//-----------------------------------------------------------------------------
+// In boost since 1.68.0
+// nvcc CUDA compiler detection
+
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 106800
```

Comment:
use `==`. It should be fixed in the next release if we provide an upstream PR now.
maybe also link the issue/PR here

Reply:
As long as we do not have an accepted solution in boost it is more future proof with `>=`. I am still thinking about the correct solution in boost/predef and this PR can be merged before this is solved.